libnetwork/drivers/bridge/bridge_linux_test.go

```diff
@@ -964,6 +966,39 @@ func TestValidateConfig(t *testing.T) {
 	}
 }
 
+func TestValidateFixedCIDRV6(t *testing.T) {
+	var err error
+
+	err = ValidateFixedCIDRV6("2001:db8::/32")
+	assert.Check(t, err == nil)
```

Comment:
I probably should've commented that yesterday during the call; these can be simplified to;

```go
assert.Check(t, err)
```

The added advantage then is that the error itself is also printed;

currently, we'd get:

```console
foo_test.go:21: assertion failed: err is not nil
```

When using `assert.Check(t, err)`, that would be;


```console
foo_test.go:21: assertion failed: error is not nil: invalid CIDR address: hello
```

Reply:
We could consider using a test-table with subtests for this; it's slightly more verbose, but may make it easier to add more test-cases in future if we see a need, and it allows running individual subtests in some IDE's. Using a "doc" (or similar) field in the table also allows describing the _intent_ of some of the test-cases (beyond just a comment);

```go
func TestValidateFixedCIDRV6(t *testing.T) {
	tests := []struct {
		doc, input, expectedErr string
	}{
		{
			doc:   "valid",
			input: "2001:db8::/32",
		},
		{
			// fixed-cidr-v6 doesn't have to be specified.
			doc:   "empty",
			input: "2001:db8::/32",
		},
		{
			// Using the default LL subnet is ok.
			doc:   "default link-local subnet",
			input: "fe80::/64",
		},
		{
			// Using an LL subnet that doesn't overlap with the default is ok.
			doc:   "non-overlapping link-local subnet",
			input: "fe80:1234::/80",
		},
		{
			// Overlapping with the default LL subnet isn't allowed.
			doc:         "overlapping link-local subnet fe80::/63",
			input:       "fe80::/63",
			expectedErr: "fixed-cidr-v6 clashes with reserved link local subnet 'fe80::/64': fe80::/63",
		},
		{
			// Overlapping with the default LL subnet isn't allowed.
			doc:         "overlapping link-local subnet fe80::/65",
			input:       "fe80::/65",
			expectedErr: "fixed-cidr-v6 clashes with reserved link local subnet 'fe80::/64': fe80::/65",
		},
		{
			// The address has to be valid IPv6 subnet.
			doc:         "invalid IPv6 subnet",
			input:       "2000:db8::",
			expectedErr: "fixed-cidr-v6 is not a valid subnet '2000:db8::'",
		},
		{
			doc:         "non-IPv6 subnet",
			input:       "10.3.4.5/24",
			expectedErr: "fixed-cidr-v6 is not an IPv6 subnet '10.3.4.5/24'",
		},
		{
			doc:         "invalid subnet",
			input:       "nonsense",
			expectedErr: "fixed-cidr-v6 is not a valid subnet 'nonsense'",
		},
	}
	for _, tc := range tests {
		tc := tc
		t.Run(tc.doc, func(t *testing.T) {
			err := ValidateFixedCIDRV6(tc.input)
			if tc.expectedErr == "" {
				assert.Check(t, err)
			} else {
				assert.Check(t, is.Error(err, tc.expectedErr))
			}
		})
	}
}
```

With that, tests would look like this;

```bash
go test -v -run TestValidateFixedCIDRV6
=== RUN   TestValidateFixedCIDRV6
=== RUN   TestValidateFixedCIDRV6/valid
=== RUN   TestValidateFixedCIDRV6/empty
=== RUN   TestValidateFixedCIDRV6/default_link-local_subnet
=== RUN   TestValidateFixedCIDRV6/non-overlapping_link-local_subnet
=== RUN   TestValidateFixedCIDRV6/overlapping_link-local_subnet_fe80::/63
=== RUN   TestValidateFixedCIDRV6/overlapping_link-local_subnet_fe80::/65
=== RUN   TestValidateFixedCIDRV6/invalid_IPv6_subnet
=== RUN   TestValidateFixedCIDRV6/non-IPv6_subnet
=== RUN   TestValidateFixedCIDRV6/invalid_subnet
--- PASS: TestValidateFixedCIDRV6 (0.00s)
    --- PASS: TestValidateFixedCIDRV6/valid (0.00s)
    --- PASS: TestValidateFixedCIDRV6/empty (0.00s)
    --- PASS: TestValidateFixedCIDRV6/default_link-local_subnet (0.00s)
    --- PASS: TestValidateFixedCIDRV6/non-overlapping_link-local_subnet (0.00s)
    --- PASS: TestValidateFixedCIDRV6/overlapping_link-local_subnet_fe80::/63 (0.00s)
    --- PASS: TestValidateFixedCIDRV6/overlapping_link-local_subnet_fe80::/65 (0.00s)
    --- PASS: TestValidateFixedCIDRV6/invalid_IPv6_subnet (0.00s)
    --- PASS: TestValidateFixedCIDRV6/non-IPv6_subnet (0.00s)
    --- PASS: TestValidateFixedCIDRV6/invalid_subnet (0.00s)
PASS
ok  	github.com/docker/docker/libnetwork/drivers/bridge	0.008s
```